seacrowd/sea_datasets/alorese/alorese.py

```diff
@@ -0,0 +1,259 @@
+                    "end_time": datasets.Value("int64"),
+                }
+            )
+
+        elif "seacrowd_sptext" in self.config.schema:
+            features = schemas.speech_text_features
+
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+        
+        else:
+            raise ValueError(f"Invalid config schema: {self.config.schema}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        if self.config.schema == "source":
+            paths = dl_manager.download(_URLS)
+        else:
+            if "t2t" in self.config.subset_id:
+                filepath = {k: v["text_path"] for k, v in _URLS.items()}
+                paths = dl_manager.download(filepath)
+            else:
+                paths = dl_manager.download(_URLS)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": paths,
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath) -> Tuple[int, Dict]:
+
+        if self.config.schema == "source":
+            source_df = self._get_source_df(filepath)
+
+            for k, row in source_df.iterrows():
+                yield k, {
+                    "nr": k + 1,
+                    "media_id": row["media_id"],
+                    "audio": row["audio_path"],
+                    "annotation_aol": row["annotation_aol"],
+                    "annotation_ind": row["annotation_ind"],
+                    "begin_time": row["begin_time"],
+                    "end_time": row["end_time"],
+                }
+
+        elif "seacrowd" in self.config.schema:
+            if "t2t" in self.config.subset_id:
+                caption_df = self._merge_text_dfs(filepath)
+
+                for k, row in caption_df.iterrows():
+                    yield k, {
+                        "id": k + 1,
+                        "text_1": row["annotation_aol"],
+                        "text_2": row["annotation_ind"],
+                        "text_1_name": _LANGUAGES[0],
+                        "text_2_name": _LANGUAGES[1],
+                    }
+            elif "sptext" in self.config.subset_id:
+                annot_lang = "annotation_aol" if self.config.subset_id.split("_")[-1] == "sptext" else "annotation_ind"
+                sptext_df = self._get_sptext_df(filepath)
+
+                for k, row in sptext_df.iterrows():
+                    yield k, {"id": k + 1, "path": row["audio_path"], "audio": row["audio_path"], "text": row[annot_lang], "speaker_id": row["speaker_id"], "metadata": {"speaker_age": None, "speaker_gender": None}}
```

Comment:
mind reformat this one so it will conform with previous elif statements?

Reply:
Sure! Please check latest push for the final nitpick 👍